test/integration/LIPs/LIP52_ClaimEarningsSnapshot.js

```diff
@@ -0,0 +1,573 @@
+import {contractId} from "../../../utils/helpers"
+import {constants} from "../../../utils/constants"
+const {MerkleTree} = require("../../../utils/merkleTree")
+import {createWinningTicket, getTicketHash} from "../../helpers/ticket"
+import signMsg from "../../helpers/signMsg"
+
+const {keccak256, bufferToHex} = require("ethereumjs-util")
+let abi = require("ethereumjs-abi")
+import BN from "bn.js"
+import truffleAssert from "truffle-assertions"
+import {assert} from "chai"
+
+const Controller = artifacts.require("Controller")
+const BondingManager = artifacts.require("BondingManager")
+const AdjustableRoundsManager = artifacts.require("AdjustableRoundsManager")
+const LivepeerToken = artifacts.require("LivepeerToken")
+const TicketBroker = artifacts.require("TicketBroker")
+const MerkleSnapshot = artifacts.require("MerkleSnapshot")
+const BondingManagerPreLIP36 = artifacts.require("BondingManagerPreLIP36")
+const LinkedList = artifacts.require("SortedDoublyLL")
+const ManagerProxy = artifacts.require("ManagerProxy")
+
+async function executeLIP36Upgrade(controller, roundsManager, bondingAddress) {
```

Comment:
Might make sense to extract this into a test helper file as the canonical LIP-36 upgrade function since it is used here as well as in `test/integration/Earnings.js`.

Reply:
fixed in b05f3fd